aerogear/unifiedpush-operator
# UPS Universal container and enmasse

This PR adds support for using AMQ Online as a message broker.  This allows UPS to run in a multitenancy fashion.

# Testing notes
Until we Publish the new UPS container, I have provided a docker formatted container that is based on the latest version of UPS.  To use this image set the ups initial image stream using `export UPS_IMAGE_STREAM_INITIAL_IMAGE=docker.io/secondsun/ups:3.0.0`

#Testing Steps

1. Run the operator with the environment variable `export UPS_IMAGE_STREAM_INITIAL_IMAGE=docker.io/secondsun/ups:3.0.0` set
2. Create a UPS server using the example as described in the readme
3. Verify the server deploys by checking the unifiedpush and AMQ Online projects for the correct resources. 
4. Create a Android or iOS variant using whatever method you feel appropriate
5. Send a message using UPS
6. Verify that the message was sent using AMQ online.  You can do this by looking at the AMQ Online console in the ups address space.  You should see on the addresses for your messaging queues that messages have been sent on those addresses
7. Undeploy UPS
8. verify that the AMQ and UPS  resources cleanup 

# Screen Shot of AMQ console showing FCM message activity 
![image](https://user-images.githubusercontent.com/2413816/61063703-6aafb000-a3ce-11e9-9218-18beca9d6111.png)
